internal/cli/template_delete.go

```diff
@@ -0,0 +1,116 @@
+package cli
+
+import (
+	"github.com/posener/complete"
+
+	"github.com/hashicorp/waypoint-plugin-sdk/terminal"
+	"github.com/hashicorp/waypoint/internal/clierrors"
+	"github.com/hashicorp/waypoint/internal/pkg/flag"
+	pb "github.com/hashicorp/waypoint/pkg/server/gen"
+)
+
+type ProjectTemplateDeleteCommand struct {
+	*baseCommand
+
+	flagName string
+	flagID   string
+}
+
+func (c *ProjectTemplateDeleteCommand) Run(args []string) int {
+	flagSet := c.Flags()
+	if err := c.Init(
+		WithArgs(args),
+		WithFlags(flagSet),
+		WithNoConfig(),
+	); err != nil {
+		return 1
+	}
+	args = flagSet.Args()
+	ctx := c.Ctx
+
+	name := ""
+	if len(args) == 1 {
+		name = args[0]
+	}
+
+	var tref pb.Ref_ProjectTemplate
+	if name != "" {
+		tref.Ref = &pb.Ref_ProjectTemplate_Name{
+			Name: name,
+		}
+	} else if c.flagID != "" {
+		tref.Ref = &pb.Ref_ProjectTemplate_Id{
+			Id: c.flagID,
+		}
+		name = c.flagID
+	} else if c.flagName != "" {
+		tref.Ref = &pb.Ref_ProjectTemplate_Name{
+			Name: c.flagName,
+		}
+		name = c.flagName
+	} else {
+		c.ui.Output("Missing project template name or id.\n\n"+c.Help(), terminal.WithErrorStyle())
+		return 1
+	}
+
+	_, err := c.project.Client().DeleteProjectTemplate(ctx, &pb.DeleteProjectTemplateRequest{
+		ProjectTemplate: &tref,
+	})
+	if err != nil {
+		c.ui.Output("Encountered an error while deleting the project template: %s", clierrors.Humanize(err), terminal.WithErrorStyle())
+		return 1
+	}
+
+	c.ui.Output("Template %q deleted", name, terminal.WithSuccessStyle())
+
+	return 0
+}
+
+func (c *ProjectTemplateDeleteCommand) Flags() *flag.Sets {
+	return c.flagSet(0, func(sets *flag.Sets) {
+		f := sets.NewSet("Command Options")
+
+		f.StringVar(&flag.StringVar{
+			Name:    "name",
+			Target:  &c.flagName,
```

Comment:
It seems strange to me that `name` is both a flag and an argument. It's also not clear to the user which would take precedence (it looks like the `name` argument does), same with `-id`

```
$ waypoint template delete -name=sonic -id=the hedgehog
```

Reply:
This is a fair point, I think it makes sense to remove the name flag and just use it as an argument. Throwing an error if both an argument and the `-id` flag are specified. That should clear up precedent ambiguity while removing unnecessary flags.